qutebrowser/browser/webengine/notification.py

```diff
@@ -361,13 +361,22 @@ def _on_adapter_clicked(self, notification_id: int) -> None:
             return
 
         try:
+            self._focus_first_matching_tab(notification)
             notification.click()
         except RuntimeError:
             # WORKAROUND for
             # https://www.riverbankcomputing.com/pipermail/pyqt/2020-May/042918.html
             log.misc.debug(f"Ignoring click request for notification {notification_id} "
                            "due to PyQt bug")
 
+    def _focus_first_matching_tab(self, notification: "QWebEngineNotification") -> None:
+        for win_id in objreg.window_registry:
+            tabbedbrowser = objreg.get("tabbed-browser", window=win_id, scope="window")
+            for idx, tab in enumerate(tabbedbrowser.widgets()):
+                if tab.url().matches(notification.origin(), QUrl.RemovePath):
+                    tabbedbrowser.widget.setCurrentIndex(idx)
+                    return
```

Comment:
Right now this just uses the first matching tab. What should happen if there are multiple matches? It might seem like buggy behavior if the wrong tab is silently focused.

Reply:
There are other options I guess. Asking the user, trying to find the more recently accessed tab, doing nothing (or something else, although I can't think of any other sensible behavior). In most cases I think this is going to be a non-issue though. At least for my use-case, this is mostly for notifications in chat apps, where having multiple tabs of the same app open doesn't really happen.